hw-1/task-2.js

```diff
@@ -0,0 +1,40 @@
+"use strict";
+/*
+Создайте функцию validatePassword, которая принимает строку (пароль) и возвращает true,
+если пароль соответствует следующим правилам:
+  - Пароль должен содержать хотя бы одну заглавную букву.
+  - Пароль должен содержать хотя бы одну букву в нижнем регистре.
+  - Пароль должен содержать хотя бы одну цифру.
+  - Пароль должен быть не менее 8 символов. +
+  - Пароль не должен состоять из одних пробелов +
+Функция должна возвращать false, если хотя бы одно из условий не выполнено.
+*/
+function validatePassword(password) {
+    let lowerCase = false;
+    let upperCase = false;
+    let number = false;
+    if (!password.length) {
+        return false;
+    }
+    else if (password.length < 8) {
+        return false;
+    }
+    else if (password.includes(' ')) {
+        return false;
+    }
+    for (let el of password) {
+        if (el >= 'a' && el <= 'z') {
+            lowerCase = true;
+        }
+        else if (el >= 'A' && el <= 'Z') {
+            upperCase = true;
+        }
+        else if (el >= '0' && el <= '9') {
+            number = true;
+        }
+        else
+            return false;
+    }
+    return upperCase === true && lowerCase === true && number === true ? true : false;
```

Comment:
Єто переменньій "предикатьі", представляют `boolean` значения, их используют без дополнительной проверки 
```suggestion
    return upperCase && lowerCase  && number;
```